### ISSUE TITLE
App crashes on screen rotation during discovering API

### ISSUE THREAD

**Issue by neupanedipen** [eduvpn/android#171]
#### Expected behavior<br>
The app must be stable on screen rotation during discovering API.<br>
#### Actual behavior<br>
But the app crashes on screen rotation during discovering API.<br>
#### How to reproduce<br>
1. Install the app from playstore.<br>
2. Launch the app.<br>
3. Click on &quot;Safely use the internet&quot; option.<br>
4. Now select aby of the country from the list. And it will start discovering the API.<br>
5. Before the discovering of API completes change the orientation of the screen to landscape mode. And click any where on the screen.<br>
The app crashes.<br>
* App Version: 1.2.2<br>
* Device: Samsung Galaxy J5<br>
* Operating System: Android 6.0.1<br>
#### Recording Of The Bug<br>
https://youtu.be/0rJGzGi36l4

**Comment by fkooman** [comment]
@mkoelewijn can you obtain the crash report from Google Play store? I try to reproduce it, but my phone from 2012 is too fast it seems...

**Comment by fkooman** [comment]
Possibly reproduced:

```
05-22 16:20:32.651 5351-5351/nl.eduvpn.app E/WindowManager: android.view.WindowLeaked: Activity nl.eduvpn.app.MainActivity has leaked window DecorView@896646e[Just a sec] that was originally added here
        at android.view.ViewRootImpl.<init>(ViewRootImpl.java:485)
        at android.view.WindowManagerGlobal.addView(WindowManagerGlobal.java:346)
        at android.view.WindowManagerImpl.addView(WindowManagerImpl.java:93)
        at android.app.Dialog.show(Dialog.java:330)
        at android.app.ProgressDialog.show(ProgressDialog.java:185)
        at android.app.ProgressDialog.show(ProgressDialog.java:142)
        at nl.eduvpn.app.fragment.ProviderSelectionFragment._connectToApi(ProviderSelectionFragment.java:204)
        at nl.eduvpn.app.fragment.ProviderSelectionFragment.access$100(ProviderSelectionFragment.java:61)
        at nl.eduvpn.app.fragment.ProviderSelectionFragment$1.onItemClicked(ProviderSelectionFragment.java:156)
        at nl.eduvpn.app.utils.ItemClickSupport$1.onClick(ItemClickSupport.java:39)
        at android.view.View.performClick(View.java:6294)
        at android.view.View$PerformClick.run(View.java:24770)
        at android.os.Handler.handleCallback(Handler.java:790)
        at android.os.Handler.dispatchMessage(Handler.java:99)
        at android.os.Looper.loop(Looper.java:164)
        at android.app.ActivityThread.main(ActivityThread.java:6494)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:438)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:807)
```

**Comment by fkooman** [comment]
If the crash report above helps you in any way and the fix is easy go for it...

**Comment by dzolnai** [comment]
Fixed with #193
